### PR TITLE
JobAbort on SFC binding (ConcurrentModificationException, issue 550)

### PR DESCRIPTION
**Pull request by dmitryintel**
* Add unit tests that replicate this exact stack trace. 
* Tested the fix. 
* [issue 550](https://github.com/opensecuritycontroller/osc-core/issues/550)

### CODE REVIEW

File: osc-server/src/main/java/org/osc/core/broker/job/Graph.java [hunk 1]
```diff
@@ -162,25 +162,37 @@ private Graph(Set<T> nodes, Multimap<T, T> edges) {
         this.edges = edges;
     }
 
-    public Set<Edge<T>> getEdges() {
+    private synchronized Set<Edge<T>> getEdges() {
         Set<Edge<T>> edgeSet = new HashSet<Edge<T>>();
-        for (Map.Entry<T, T> edge : this.edges.entries()) {
+
+        Set<Map.Entry<T, T>> copy;
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/job/Graph.java]: you dont neeed this since the method is already synchronized

Comment (dmitryintel) [reply osc-server/src/main/java/org/osc/core/broker/job/Graph.java]: I actually thought of removing synchronized.

File: osc-server/src/main/java/org/osc/core/broker/job/Graph.java [hunk 2]
```diff
@@ -412,13 +406,12 @@ public boolean equals(Object obj) {
 
         @SuppressWarnings("unchecked")
         Graph<T> that = (Graph<T>) obj;
-        return new EqualsBuilder().append(this.nodes, that.nodes).append(this.edges, that.edges).isEquals();
+        return new EqualsBuilder().append(getNodes(), getNodes()).append(getEdges(), getEdges()).isEquals();
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/job/Graph.java]: should be that.getNodes

File: osc-server/src/test/java/org/osc/core/broker/job/JobEngineTest.java [hunk 3]
```diff
@@ -243,6 +246,89 @@ public void testTaskDependencyExecutionOrder() throws Exception {
         verifyJobPersistence(this.job);
     }
 
+    @Test
+    public void testConcurrentGraphModification() throws Exception {
```
Comment (arvindn05) [inline osc-server/src/test/java/org/osc/core/broker/job/JobEngineTest.java]: add defect number for more detials

**Pull request merged**